tedana/metrics/dependence.py

```diff
@@ -0,0 +1,354 @@
+             signal_FT2_Z, noise_FT2_Z, equal_var=False)
+
+    signal_minus_noise_t[np.isnan(signal_minus_noise_t)] = 0
+    signal_minus_noise_p[np.isnan(signal_minus_noise_p)] = 0
+    return signal_minus_noise_t, signal_minus_noise_p
+
+
+def compute_dice(Br_clmaps, F_clmaps):
+    n_components = Br_clmaps.shape[1]
+    dice_values = np.zeros(n_components)
+    for i_comp in range(n_components):
+        dice_values[i_comp] = utils.dice(Br_clmaps[:, i_comp], F_clmaps[:, i_comp])
+
+    dice_values[np.isnan(dice_values)] = 0
+    return dice_values
+
+
+def generate_decision_table_score(kappa, dice_FT2, signal_minus_noise_t,
+                                  countnoise, countsigFT2):
+    d_table_rank = np.vstack([
+        len(kappa) - stats.rankdata(kappa),
+        len(kappa) - stats.rankdata(dice_FT2),
+        len(kappa) - stats.rankdata(signal_minus_noise_t),
+        stats.rankdata(countnoise),
+        len(kappa) - stats.rankdata(countsigFT2)]).T
+    d_table_score = d_table_rank.mean(axis=1)
+    return d_table_score
+
+
+def calculate_z_maps(weights, z_max=8):
+    n_components = weights.shape[1]
+    Z_maps = np.zeros(weights.shape)
+    for i_comp in range(n_components):
+        # compute weights as Z-values
+        weights_z = (weights[:, i_comp] - weights[:, i_comp].mean()) / weights[:, i_comp].std()
+        weights_z[np.abs(weights_z) > z_max] = (z_max * (np.abs(weights_z) / weights_z))[
+            np.abs(weights_z) > z_max]
+        Z_maps[:, i_comp] = weights_z
+    return Z_maps
+
+
+def calculate_dependence_metrics(comptable, F_T2_maps, F_S0_maps, Z_maps):
+    RepLGR.info('Kappa (kappa) and Rho (rho) were calculated as measures of '
+                'TE-dependence and TE-independence, respectively.')
+    _names = ['kappa', 'rho']
+    if any([name in comptable.columns for name in _names]):
+        raise Exception('Metrics already exist in component table.')
+
+    n_components = Z_maps.shape[1]
+    kappas, rhos = np.zeros(n_components), np.zeros(n_components)
+    for i_comp in range(n_components):
+        norm_weights = np.abs(Z_maps[:, i_comp] ** 2.)
+        kappas[i_comp] = np.average(F_T2_maps[:, i_comp], weights=norm_weights)
+        rhos[i_comp] = np.average(F_S0_maps[:, i_comp], weights=norm_weights)
+    comptable['kappa'] = kappas
+    comptable['rho'] = rhos
+    return comptable
+
+
+def calculate_varex(optcom_betas):
+    n_components = optcom_betas.shape[1]
+    totvar = (optcom_betas**2).sum()
+
+    varex = np.zeros(n_components)
+    for i_comp in range(n_components):
+        varex[i_comp] = 100 * (optcom_betas[:, i_comp]**2).sum() / totvar
+    return varex
+
+
+def calculate_varex_norm(weights):
+    n_components = weights.shape[1]
+    totvar_norm = (weights**2).sum()
+
+    varex_norm = np.zeros(n_components)
+    for i_comp in range(n_components):
+        varex_norm[i_comp] = (weights[:, i_comp]**2).sum() / totvar_norm
+    return varex_norm
```

Comment:
This is the same as `calculate_varex()` but just doesn't multiply by 100? We should consolidate these functions.

Reply:
I'll clean it up like `calculate_varex()`, but I'd rather wait on merging until I believe in the variance explained calculations a bit more (per #317).